pkg/volume/util/util.go

```diff
@@ -192,3 +193,38 @@ func CheckNodeAffinity(pv *v1.PersistentVolume, nodeLabels map[string]string) er
 	}
 	return nil
 }
+
+// ValidateLocalPath validates a single path, returning an error if the path is
+// invalid. Paths may not:
+//
+// 1. be relative
+// 2. contain '..' as an element
+// 3. start with '..'
+// 4. contain filename larger than 255 characters
+// 5. be longer than 4096 characters
+func ValidateLocalPath(targetPath string) error {
+	if targetPath == "" {
+		return fmt.Errorf("invalid path: must not be empty: %q", targetPath)
+	}
+	if !path.IsAbs(targetPath) {
+		return fmt.Errorf("invalid path: must be absolute path: %s", targetPath)
+	}
+	if len(targetPath) > maxPathLength {
+		return fmt.Errorf("invalid path: must be less than %d characters", maxPathLength)
+	}
+
+	items := strings.Split(targetPath, string(os.PathSeparator))
+	for _, item := range items {
+		if item == ".." {
+			return fmt.Errorf("invalid path: must not contain '..': %s", targetPath)
+		}
+		if len(item) > maxFileNameLength {
```

Comment:
Since there are differences between platforms, I don't think we should be enforcing this this way. Let's limit this to the backstep path segment prevention for now